src/porepy/examples/tracer_flow.py

```diff
@@ -0,0 +1,274 @@
+
+
+class TracerBC(BoundaryConditionsMulticomponent):
+    """Boundary conditions for pressure, flow and tracer.
+
+    Mixes in the BC for pressure and the boundary type definition, and inherits the
+    BC treatment for multi-component fluids.
+
+    """
+
+    def bc_type_darcy_flux(self, sd: pp.Grid) -> pp.BoundaryCondition:
+        """Flagging the inlet and outlet faces as Dirichlet boundary, where pressure
+        is given."""
+        dirichlet_faces = np.zeros(sd.num_faces, dtype=bool)
+        # Define boundary faces on grids which are not points:
+        if sd.dim > 0:
+            sides = self.domain_boundary_sides(sd)
+            # need to cast, bg exists if sd is not point
+            bg = cast(pp.BoundaryGrid, self.mdg.subdomain_to_boundary_grid(sd))
+            bg_sides = self.domain_boundary_sides(bg)
+            inlet = inlet_faces(bg, bg_sides)
+            outlet = outlet_faces(bg, bg_sides)
+
+            dirichlet = np.zeros(bg.num_cells, dtype=bool)
+            dirichlet[inlet | outlet] = True
+
+            # broadcast to proper size
+            dirichlet_faces[sides.all_bf] = dirichlet
+
+        return pp.BoundaryCondition(sd, dirichlet_faces, "dir")
+
+    def bc_type_fluid_flux(self, sd: pp.Grid) -> pp.BoundaryCondition:
+        """Returns for the upwinding discretization the same inlet and outlet faces
+        marked as 'dir' as for the elliptic discretization."""
+        return self.bc_type_darcy_flux(sd)
+
+    def bc_values_pressure(self, bg: pp.BoundaryGrid) -> np.ndarray:
+        """Defines some non-trivial values on inlet and outlet faces of the matrix."""
+
+        p = np.zeros(bg.num_cells)
+
+        # defining BC values only on matrix
+        if bg.parent.dim == 2:
+            sides = self.domain_boundary_sides(bg)
+            inlet = inlet_faces(bg, sides)
+            outlet = outlet_faces(bg, sides)
+
+            p[inlet] = 1.5
+            p[outlet] = 1.0
+
+        return p
+
+    def bc_values_overall_fraction(
+        self, component: pp.Component, bg: pp.BoundaryGrid
+    ) -> np.ndarray:
+        """Defines some non-trivial inflow of the tracer component on the inlet."""
+
+        z = np.zeros(bg.num_cells)
+
+        assert component.name == "tracer", "Only the tracer is independent."
+
+        # 10% tracer inflow into matrix
+        if bg.parent.dim == 2:
+            sides = self.domain_boundary_sides(bg)
+            inlet = inlet_faces(bg, sides)
+
+            z[inlet] = 0.1
+
+        return z
+
+
+class TracerFlowSetup(  # type: ignore[misc]
+    DomainX,
+    TracerFluid,
+    CompositionalVariables,
+    ComponentMassBalanceEquations,
+    TracerBC,
+    TracerIC,
+    SinglePhaseFlow,
+):
+    """Complete set-up for tracer flow modelled as a single phase, 2-component flow
+    problem."""
+
+
+# If executed as main, run simulation
+if __name__ == "__main__":
+    # initial time step 60 seconds
+    dt_init = 60
+    # Simulation time 2 hour
```

Comment:
FYI, we have pp.HOUR defined in common_constants.py. I don't insist on changing here, to be clear.

Reply:
That's nice, switched to `pp.Minutes` and `pp.Hours`.